File: LeetCode/83. Remove Duplicates from Sorted List/elixir/solution.exs
# Definition for singly-linked list.
#
# defmodule ListNode do
#   @type t :: %__MODULE__{
#           val: integer,
#           next: ListNode.t() | nil
#         }
#   defstruct val: 0, next: nil
# end

defmodule Solution do
  @spec delete_duplicates(head :: ListNode.t() | nil) :: ListNode.t() | nil
  def delete_duplicates(head) do
    head
    |> node_to_list()
    |> Enum.uniq()
    |> list_to_node()
  end

  defp node_to_list(node, acc \\ [])
  defp node_to_list(nil, acc), do: acc
  defp node_to_list(node, acc), do: node_to_list(node.next, [node.val | acc])

  defp list_to_node(list, acc \\ nil)
  defp list_to_node([], acc), do: acc
  defp list_to_node([h | t], acc), do: list_to_node(t, %ListNode{val: h, next: acc})
end

File: LeetCode/83. Remove Duplicates from Sorted List/java/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode node = head;
        while (node != null) {
            ListNode next = node.next;
            while (next != null && node.val == next.val) {
                next = next.next;
            }
            node.next = next;
            node = node.next;
        }
        return head;
    }
}
